tests/integration/test_run_int.py

```diff
@@ -13,7 +13,11 @@
 @pytest.mark.skipif(IS_PYPY, reason="setuptools distutil1s patching does not work")
 def test_app_data_pinning(tmp_path):
     version = "19.1.1" if sys.version_info[0:2] == (3, 4) else "19.3.1"
-    result = cli_run([ensure_text(str(tmp_path)), "--pip", version, "--activators", "", "--seeder", "app-data"])
+    with pytest.raises(SystemExit):
+        cli_run(["--upgrade-embed-wheels"])
```

Comment:
This is quite a heavy way to get things tested but is easy to do.
Another way would be to add unit tests for `from_bundle` but this might take me much more time to implement.

Reply:
> Another way would be to add unit tests for `from_bundle` but this might take me much more time to implement.

I'd strongly prefer this.